utility-plane-services/policy-engine-api/src/main/java/org/opendatamesh/platform/up/policy/api/v1/clients/PolicyEngineClientImpl.java

```diff
@@ -0,0 +1,84 @@
+package org.opendatamesh.platform.up.policy.api.v1.clients;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import org.opendatamesh.platform.core.commons.clients.ODMClient;
+import org.opendatamesh.platform.core.commons.clients.resources.ErrorRes;
+import org.opendatamesh.platform.core.commons.servers.exceptions.BadRequestException;
+import org.opendatamesh.platform.core.commons.servers.exceptions.InternalServerException;
+import org.opendatamesh.platform.core.dpds.ObjectMapperFactory;
+import org.opendatamesh.platform.up.policy.api.v1.resources.DocumentResource;
+import org.opendatamesh.platform.up.policy.api.v1.resources.EvaluationResource;
+import org.opendatamesh.platform.up.policy.api.v1.resources.errors.PolicyEngineApiStandardErrors;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpMethod;
+import org.springframework.http.ResponseEntity;
+
+public class PolicyEngineClientImpl extends ODMClient implements PolicyEngineClient {
+
+    public PolicyEngineClientImpl(String serverAddress, ObjectMapper mapper) {
+        super(serverAddress, mapper);
+    }
+
+    public PolicyEngineClientImpl(String serverAddress) {
+        super(serverAddress, ObjectMapperFactory.JSON_MAPPER);
+    }
+
+
+    // ======================================================================================
+    // Evaluate Document
+    // ======================================================================================
+
+    public EvaluationResource evaluateDocument(DocumentResource document) {
+        ResponseEntity<ObjectNode> evaluationResponse = evaluateDocumentResponseEntity(document);
+        return mapResponseOrThrowError(evaluationResponse);
```

Comment:
su restUtils ci dovrebbero essere dei metodi di supporto proprio per questa roba qua

Reply:
Ho preso spunto da quelle infatti, ma generalizzarlo come utils comuni al momento era meno semplice del previsto per la gestione dei codici di errore

Reply:
mettiamo in backlog il refactor di RestUtils e dei suoi utilizzi, ora non credo che abbiamo tempo per farlo

Reply:
okok segna entrambi